File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.py
import sys

n, k = map(int, sys.stdin.readline().split())
arr = list(range(1, n+1))
ans = []
cnt = 0

for i in range(n):
    cnt += k-1
    if cnt >= len(arr):
        cnt %= len(arr)

    ans.append(arr.pop(cnt))

sys.stdout.write("<" + ", ".join(map(str, ans)) + ">\n")

File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.cc
#include <iostream>
#include <vector>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);

    int n, k;
    vector<int> arr, ans;
    int cnt = 0;
    cin >> n >> k;

    for (int i = 1; i <= n; i++)
        arr.push_back(i);

    for (int i = 0; i < n; i++) {
        cnt += k-1;

        if (cnt >= arr.size())
            cnt %= arr.size();

        ans.push_back(arr[cnt]);
        arr.erase(arr.begin() + cnt);
    }

    cout << "<" << ans[0];
    for (int i = 1; i < n; i++)
        cout << ", " << ans[i];
    cout << ">" << "\n";
    
    return 0;
}